### ISSUE TITLE
Joi.extend base won't compile

### ISSUE THREAD

**Issue by typeofweb** [hjkcai/typesafe-joi#14]
Code to reproduce:

```ts
const x = Joi.extend({
  base: Joi.array(),
})
```

fails with the following error:

```
Argument of type '{ base: ArraySchema<Value<any[], never, never, undefined, never>>; }' is not assignable to parameter of type 'Extension | Extension[]'.
  Property 'name' is missing in type '{ base: ArraySchema<Value<any[], never, never, undefined, never>>; }' but required in type 'Extension'. ts(2345)
```

This is likely because `extends` is defined with the use of `JoiLib.Extension` which points to the original `@types/hapi__joi` typings, and `Joi.array` defined there is not compatible with the one from `typesafe-joi`.

**Comment by hjkcai** [comment]
This error was because you forgot `name` option in the `Extension` type. However, the `Extension` itself should be adapted to typesafe-joi.

Fixed in 2.0.2.

**Comment by typeofweb** [comment]
Oh, my full code did have `name`, I accidentally must've deleted it and pasted the wrong error 🤦‍♂  Sorry about that.
Thanks for the quick fix! 👍
